ghcide/src/Development/IDE/Core/Debouncer.hs

```diff
@@ -40,17 +41,18 @@ newAsyncDebouncer = Debouncer . asyncRegisterEvent <$> newVar Map.empty
 -- to mask if required.
 asyncRegisterEvent :: (Eq k, Hashable k) => Var (HashMap k (Async ())) -> Seconds -> k -> IO () -> IO ()
 asyncRegisterEvent d 0 k fire = do
-    modifyVar_ d $ \m -> mask_ $ do
-        whenJust (Map.lookup k m) cancel
```

Comment:
wouldn't a `forkIO cancel` also work, and possibly be clearer?

Reply:
it's not equivalent. `forkIO` will create a new thread, just to throw an exception to another thread. I'm not convinced it's a good idea